lib/scanner.l

```diff
@@ -82,7 +82,7 @@ include_open ^[ \t]*@include[ \t]+\"
 <STRING>\\t       { scanctx_append_char(yyextra, '\t'); }
 <STRING>\\f       { scanctx_append_char(yyextra, '\f'); }
 <STRING>\\\\      { scanctx_append_char(yyextra, '\\'); }
-<STRING>\\\"      { scanctx_append_char(yyextra, '\"'); }
+<STRING>\\\"      { scanctx_append_string(yyextra, "\""); }
```

Comment:
Why is this change necessary? It does the same thing either way.

Reply:
The new code appends two characters: `\` and `"`. The old code appends a single character: `\"`, which is an escaped double-quote. Subtle difference, but see the test for proof.

Reply:
Which compiler are you using? "\"" should be interpreted as a string of length one, containing a double quote character. For example with gcc:

int main(void) {
  char c = '\"';
  const char *s = "\"";
  printf("%c\n", c);
  printf("%s\n", s);
  return 0;
}

the output is:

"
"

If you wanted to append \" it would have to be "\\\""

Reply:
Ugh... the HTML unescaped my escapes...added spaces to make it clear:

char c = " \ ' ";
const char *s = " \ " ";

Reply:
The difference is that one is a string literal while the other is a char literal. A string literal has the implicit trailing null.

Reply:
I misspoke before. The two characters being appended in the new code are in fact `"` and `NUL`.

Reply:
Ah, I see now. The problem is in strbuf_append_char(). It needs to append a NUL byte after appending the character.

void strbuf_append_char(strbuf_t *buf, char c)
{
  strbuf_ensure_capacity(buf, 1);
  *(buf->string + buf->length) = c;
  ++(buf->length);
  *(buf->string + buf->length) = NUL;  // Add this.
}

Then you can revert the change in scanner.l

Reply:
Ah, that works. Thanks. Interested in me updating the pull request to reflect the change?

Reply:
Yes please. Also could you expand the new test to check the other escaped chars that were appended using strbuf_append_char(), which have the same problem. For example strings ending in a backslash.

Reply:
Will do.